### PR TITLE
Application: Slothunter

### PR DESCRIPTION
**Pull request by AurevoirXavier**
#### Project Abstract

I noticed nobody realized there was a slot action, and no one participated in it until Kusama's 58th auction ended.

This could help small teams to catch the chance.

And this also could help teams/users to bid/contribute automatically.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] A BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (aUSD/USDT) address for the payment of the milestones is provided in the application.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by keeganquigley — CHANGES_REQUESTED** [review]
Thanks for the application @AurevoirXavier just a few initial questions/comments for you:

- Could you please add the technical specifics to the delivery table? These are the requirements of the contract, so each section should contain concrete technical deliverables that we can evaluate (rather than referring to the overview).
- In my personal opinion, the price is too high for one month, to design a small CLI app. Would you be willing to reduce the price and perhaps apply for a follow-up grant once your PoC is established? 
- Have you done any "temperature checks" in the community to see if this is a needed app? For example presenting your idea in forums, Discord, etc. to see if people would use it?
- Could you also flesh out the technical specifics more in the overview? For example, what language are you planning to write the app in? Can you define the functions/methods a bit? You mentioned webhooks, but how do you plan on notifying users? How does the "autobid" function work, etc.?

File: applications/slothunter.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+### Team LinkedIn Profiles (if available)
+
+- https://www.linkedin.com/in/xavier-lau-b0655310a
+
+### Overview
+
+- **Total Estimated Duration:** 5 weeks
+- **Full-Time Equivalent (FTE):**  1 FTE
+- **Total Costs:** 18,000 USD in Ethereum USDC
```
Comment (keeganquigley) [inline applications/slothunter.md:94]:
```suggestion
- **Total Costs:** 18,000 USD
```
Can you remove this to ensure that it gets parsed correctly by our GH actions?

File: applications/slothunter.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
+# Subalfred
+
+- **Team Name:** hack-ink
+- **Payment Address:** 0xccd79b2b77e71292924e85de130bfb1c03d94522 (Ethereum)
```
Comment (keeganquigley) [inline applications/slothunter.md:4]:
```suggestion
- **Payment Address:** 0xccd79b2b77e71292924e85de130bfb1c03d94522 (Ethereum) USDC
```

Comment (keeganquigley) [reply applications/slothunter.md:4]: You can add it here at the top instead.

File: applications/slothunter.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
+# Subalfred
```
Comment (semuelle) [inline applications/slothunter.md:1]:
```suggestion
# Slothunter
```

**Review by semuelle — CHANGES_REQUESTED** [review]
I agree with everything Keegan said. Also, would it be possible to configure the structure of the output, so it could be more easily processed by other programs (e.g., stdout, XML, JSON, etc.)?

**Comment by randombishop** [comment]
Hi @AurevoirXavier! 
Thanks for this application.
One question from my end: would you consider reducing the price under Level 1 given the scope and difficulty of the new features?
Thank you!

**Comment by AurevoirXavier** [comment]
@keeganquigley @semuelle @randombishop 

Hi guys!

I made some updates. Please feel free to tell me what I can improve more.

Thanks so much.

**Comment by keeganquigley** [comment]
thanks for the changes @AurevoirXavier, it's looking better, but can you still integrate more technical details into the delivery boxes itself, since these are the requirements that will be evaluated. Or at least add a link to the sections you are referring to.  The delivery tables for M1 & M2 in your [Subalfred application](https://github.com/w3f/Grants-Program/blob/master/applications/subalfred.md) looked great, for example.

Also, one follow-up question: Would the users be able to select which emails/webhooks they want to receive? You mentioned that some emails might be annoying, but perhaps the user should have the choice to decide this themselves?

**Comment by AurevoirXavier** [comment]
> thanks for the changes @AurevoirXavier, it's looking better, but can you still integrate more technical details into the delivery boxes itself, since these are the requirements that will be evaluated. Or at least add a link to the sections you are referring to. The delivery tables for M1 & M2 in your [Subalfred application](https://github.com/w3f/Grants-Program/blob/master/applications/subalfred.md) looked great, for example.

Thanks for the quick response. How about now?

> Also, one follow-up question: Would the users be able to select which emails/webhooks they want to receive? You mentioned that some emails might be annoying, but perhaps the user should have the choice to decide this themselves?

The email and webhook addresses are configurable.

Thanks for your suggestion. I made some changes, users should control these themselves.

**Comment by keeganquigley** [comment]
thanks @AurevoirXavier one last minor issue - If M1 is 4 weeks and M2 is 2 weeks, I think the **Total Estimated Duration** should be 6 weeks instead of 5.

**Comment by AurevoirXavier** [comment]
> thanks @AurevoirXavier one last minor issue - If M1 is 4 weeks and M2 is 2 weeks, I think the **Total Estimated Duration** should be 6 weeks instead of 5.

Thanks, I have just updated it now.

**Comment by keeganquigley** [comment]
@AurevoirXavier thanks for the updates. I will go ahead and mark your application as ready for review and share it internally with the rest of the committee so that others can comment.

Regarding M1, I think you could still add more technical specifications to the table, such as functions/methods that you plan to write, and more details on how the components will work/what exactly is being built.

Regarding M2, we usually don't fund purely educational materials, but if it involves additional coding to go with the tutorial, I think this should be fine.

**Review by semuelle — APPROVED** [review]
Thanks for the updates. Happy to proceed!

**Review by randombishop — APPROVED** [review]
OK

**Review by keeganquigley — APPROVED** [review]

**Comment by AurevoirXavier** [comment]
> Hi @AurevoirXavier Can you provide an update on Milestone 1?

Sure!

I was struggling with the Substrate API Rust client, such as subxt or substrate-api-client.

And I decided to implement a new one that is easier to use. After I spent some time on that, I realize that I can't have 2 grants at the same time.

So, I was back to this a few days ago!

I'm sorry about the delay!

**Comment by keeganquigley** [comment]
Great @AurevoirXavier thanks for the update!

**Pull request merged**